### PR TITLE
Use email as primary identifier, and introduce proper future-proof "slugging" for users

### PR DESCRIPTION
**Pull request by magnuhar**
### FEAT: Use email as primary identifier (remove username), and introduce proper future-proof "slugging" for users

This PR makes a few drastic, but important, changes to the Users app. The practical result is that username has been removed, and is now no longer used for logging in/representational values or slugs. Instead we use emails as the primary identifier. Slugs, which used to be based on usernames, now base themselves on the name of the user. It is also possible to edit personalia from the my profile section now.

Due to how the data structure is designed I had to some unconventional hacks or work-arounds. Ideally these should be solved in the future, but I did not feel as if they were in the scope of this PR (which has already exploded quite a bit). The core issue revolves around us having a User model (in User app), and a Person model in Webook app. We can not store user personalia on the user, as that would not be DRY, but we need the personalia contained in the person to get a proper slug for the user.
This in practice means;
1. We need to create a person when a user is created
2. We need to be able to generate a slug even if a person does not exist
3. A user needs to inform us of his/her personalia while registering, meaning we have to register two instances in the signup process (User and Person, between which a relationship must be established).

We want our slugs for the users to base themselves on the names of the users. 
For instance;
`wemade.no/users/john_smith`

**Concrete**
- Added First Name / Middle Name / Last Name fields to register user form
- When registering a user, a Person and a User will be established, based on the values in the signup form. 
- Reconfigured AllAuth. Use email as primary identifier, hide username. Use custom signup form.
- Extend Users model, so as to be able to get a slug in a dynamic fashion. We should be able to still create a slug, even if there is no person attached to the User. If a person is attached, use his/her name, if not use Users' guaranteed email. 

**The issues**
To get all of this done I had to create a form for the Person model. This form currently exists in users, and in a users template. This is very icky, but there is no other place to put this just yet. Ideally i think this should be imported. We'll probably be seeing this same import being used for contact persons on Organizations/Arrangements/Events...
The solution should be so that Users relies on WeBook, but WeBook should never depend or know about Users (except for the more abstract interfaces provided by Django)

### CODE REVIEW

File: webook/users/forms.py [hunk 1]
```diff
@@ -1,36 +1,38 @@
...
-            self.error_messages["duplicate_username"]
-        )
+class UserCreationForm(SignupForm):
+
+    first_name = dj_forms.CharField(max_length=512, label=_("First Name"))
+    middle_name = dj_forms.CharField(max_length=512, label=_("Middle Name"))
+    last_name = dj_forms.CharField(max_length=512, label=_("Last Name"))
+    
+    def save (self, request):
+        user = super(UserCreationForm, self).save(request)
```
Comment (eivl) [inline webook/users/forms.py:26]: python 2 syntax is not needed when calling super() anymore.

**Review by eivl — APPROVED** [review]
It does feel a bit overengineered, and Im not sure it is that future proof, but I do hope I'm wrong about that!

**Comment by eivl** [comment]
@magnuhar lets look at the conflicts together.

**Comment by magnuhar** [comment]
I'm not sure myself about the complexity of this, to be honest.
I'm going to do this again for the other Django project this evening, so I should be able to tell after the second go if this complexity is necessary.

**Pull request merged**